h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java

```diff
@@ -168,40 +172,40 @@ public XGBoostModel(Key<XGBoostModel> selfKey, XGBoostParameters parms, XGBoostO
     }
     if (p._eta != 0.3) {
       Log.info("Using user-provided parameter eta instead of learn_rate.");
-      params.put("eta", p._eta);
+      params.put("eta", localizedNumberFormatter.format(p._eta));
```

Comment:
Instead of handling every single case I would recommend allowing to put the number directly in the map and then post-process it to a correct XGBoost input. This will prevent future mistakes.

Reply:
👍 to this, someone might just add a new param as is without realizing this in the future. A post process of all float/double values right before we return the params map might be safer.

Reply:
Fixed.